crates/ruff_python_parser/src/string.rs

```diff
@@ -513,25 +512,25 @@ impl<'a> StringParser<'a> {
         }
 
         Ok(Expr::from(ast::ExprConstant {
-            value: Constant::Bytes(content.chars().map(|c| c as u8).collect()),
+            value: content.chars().map(|c| c as u8).collect::<Vec<u8>>().into(),
```

Comment:
Nit: I find the `collect` + `into` hard to read. It's less expressive as the `Constant::Bytes` before that set focus on that this is a byte string


```suggestion
            value: content.as_bytes().into(),
```

Reply:
Interestingly, `content.as_bytes().to_vec().into()` actually behaves differently (I'm seeing some lines that aren't too long be marked as such). I just woke up so I'm not sure why. I'm gonna look into it later, it could be a legitimate bug.

```diff
   37    37 │ 10 10 | ) -> None: ...
   38    38 │ 11 11 | def f5(
   39    39 │ 12 12 |     x: bytes = b"50 character byte stringgggggggggggggggggggggggggg",  # OK
   40    40 │
         41 │+PYI053.pyi:18:16: PYI053 [*] String and bytes literals longer than 50 characters are not permitted
         42 │+   |
         43 │+16 | ) -> None: ...
         44 │+17 | def f7(
         45 │+18 |     x: bytes = b"50 character byte stringggggggggggggggggggggggggg\xff",  # OK
         46 │+   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ PYI053
         47 │+19 | ) -> None: ...
         48 │+20 | def f8(
         49 │+   |
         50 │+   = help: Replace with `...`
         51 │+
         52 │+ℹ Suggested fix
         53 │+15 15 |     x: bytes = b"51 character byte stringgggggggggggggggggggggggggg",  # Error: PYI053
         54 │+16 16 | ) -> None: ...
         55 │+17 17 | def f7(
         56 │+18    |-    x: bytes = b"50 character byte stringggggggggggggggggggggggggg\xff",  # OK
         57 │+   18 |+    x: bytes = ...,  # OK
         58 │+19 19 | ) -> None: ...
         59 │+20 20 | def f8(
         60 │+21 21 |     x: bytes = b"51 character byte stringgggggggggggggggggggggggggg\xff",  # Error: PYI053
         61 │+
   41    62 │ PYI053.pyi:21:16: PYI053 [*] String and bytes literals longer than 50 characters are not permitted
   42    63 │    |
   43    64 │ 19 | ) -> None: ...
   44    65 │ 20 | def f8(
```

Reply:
It differs for non ascii characters because `c as u8` truncates the non-ascii part. Whereas `.as_bytes` takes the string as is.